soc/arm/st_stm32/stm32h7/power_m7.c

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright (c) 2021 SILA Embedded Solutions GmbH <office@embedded-solutions.at>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <zephyr.h>
+#include <pm/pm.h>
+#include <soc.h>
+#include <init.h>
+
+#include <stm32h7xx_ll_pwr.h>
+
+#include <logging/log.h>
+LOG_MODULE_DECLARE(soc, CONFIG_SOC_LOG_LEVEL);
+
+/* Initialize STM32 Power */
+static int stm32_power_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+#if !defined(SMPS) && \
+		(defined(CONFIG_POWER_SUPPLY_DIRECT_SMPS) || \
+		defined(CONFIG_POWER_SUPPLY_SMPS_1V8_SUPPLIES_LDO) || \
+		defined(CONFIG_POWER_SUPPLY_SMPS_2V5_SUPPLIES_LDO) || \
+		defined(CONFIG_POWER_SUPPLY_SMPS_1V8_SUPPLIES_EXT_AND_LDO) || \
+		defined(CONFIG_POWER_SUPPLY_SMPS_2V5_SUPPLIES_EXT_AND_LDO) || \
+		defined(CONFIG_POWER_SUPPLY_SMPS_1V8_SUPPLIES_EXT) || \
+		defined(CONFIG_POWER_SUPPLY_SMPS_2V5_SUPPLIES_EXT))
+#error Unsupported configuration: Selected SoC do not support SMPS
+#endif
+#if defined(CONFIG_POWER_SUPPLY_DIRECT_SMPS)
+	LL_PWR_ConfigSupply(LL_PWR_DIRECT_SMPS_SUPPLY);
+#elif defined(CONFIG_POWER_SUPPLY_SMPS_1V8_SUPPLIES_LDO)
+	LL_PWR_ConfigSupply(LL_PWR_SMPS_1V8_SUPPLIES_LDO);
+#elif defined(CONFIG_POWER_SUPPLY_SMPS_2V5_SUPPLIES_LDO)
```

Comment:
While you're at it, would you mind using `IS_ENABLED` instead of `#ifdef's` for readability ?
```suggestion
	if (IS_ENABLED(CONFIG_POWER_SUPPLY_DIRECT_SMPS)) {
		LL_PWR_ConfigSupply(LL_PWR_DIRECT_SMPS_SUPPLY);
	} else if (IS_ENABLED(CONFIG_POWER_SUPPLY_SMPS_1V8_SUPPLIES_LDO)) {
		LL_PWR_ConfigSupply(LL_PWR_SMPS_1V8_SUPPLIES_LDO);
	.....
```
Would be nice...

Reply:
done

Reply:
Unfortunately I had to switch back to the #if defined approach, as otherwise the code would not build for some platforms. For some platforms the symbol LL_PWR_DIRECT_SMPS_SUPPLY for instance might not be defined.